hikari/api/special_endpoints.py

```diff
@@ -1887,95 +1865,199 @@ def add_component(self, component: ComponentBuilder, /) -> Self:
             The builder object to enable chained calls.
         """
 
-    @typing.overload
     @abc.abstractmethod
-    def add_button(
-        self, style: components_.InteractiveButtonTypesT, custom_id: str, /
-    ) -> InteractiveButtonBuilder[Self]:
-        ...
+    def add_interactive_button(
+        self,
+        style: components_.InteractiveButtonTypesT,
+        custom_id: str,
+        /,
+        *,
+        emoji: typing.Union[snowflakes.Snowflakeish, emojis.Emoji, str, undefined.UndefinedType] = undefined.UNDEFINED,
+        label: undefined.UndefinedOr[str] = undefined.UNDEFINED,
+        is_disabled: bool = False,
+    ) -> Self:
+        """Add an interactive button component to this action row builder.
 
-    @typing.overload
-    @abc.abstractmethod
-    def add_button(
-        self, style: typing.Literal[components_.ButtonStyle.LINK, 5], url: str, /
-    ) -> LinkButtonBuilder[Self]:
-        ...
+        Either `emoji` xor `label` must be provided to be the button's
```

Comment:
```suggestion
        Either `emoji` or `label` must be provided to be the button's
```

Reply:
that's an exclusive or

Reply:
Would an easier language be: `Exactly one of `emoji` or `label` must be provided to be the button's`?

Reply:
I just went with the common dev/logic term cause its concise. Otherwise I would prob phrase it like ``Either `emoji` or `label` must be provided to be the button's displayed label, but not both.`` but that's not as concise.

Reply:
`xor` might be more confusing, so I would rather keep it more "english"